nrpy/equations/nrpyelliptic/ConformallyFlat_RHSs.py

```diff
@@ -0,0 +1,161 @@
+        #          Derivative variables like these must have an underscore
+        #          in them, so the finite difference module can parse the
+        #          variable name properly.
+        uu_dD = ixp.declarerank1("uu_dD")
+
+        # Step 4b: Declare the rank-2 indexed expression \partial_{ij} u,
+        #          which is symmetric about interchange of indices i and j
+        #          Derivative variables like these must have an underscore
+        #          in them, so the finite difference module can parse the
+        #          variable name properly.
+        uu_dDD = ixp.declarerank2("uu_dDD", symmetry="sym01")
+
+        # Step 5.a: Declare spatially-dependent variable_wavespeed as grid function
+        if "variable_wavespeed" not in gri.glb_gridfcs_dict:
+            variable_wavespeed = gri.register_gridfunctions(
+                "variable_wavespeed", group="AUXEVOL", gf_array_name="auxevol_gfs"
+            )[0]
+        else:
+            variable_wavespeed = sp.Symbol("variable_wavespeed", real=True)
+
+        # Step 5.b: Register AUXEVOL gridfunctions that are part of RHSs
+        if "psi_background" not in gri.glb_gridfcs_dict:
+            psi_background, ADD_times_AUU = gri.register_gridfunctions(
+                ["psi_background", "ADD_times_AUU"],
+                group="AUXEVOL",
+                gf_array_name="auxevol_gfs",
+            )
+        else:
+            psi_background, ADD_times_AUU = sp.symbols(
+                "psi_background ADD_times_AUU", real=True
+            )
+
+        # Step 5.c: Register residual as AUX gridfunction
+        if "residual_H" not in gri.glb_gridfcs_dict:
+            _residual_H = gri.register_gridfunctions(
+                "residual_H", group="AUX", gf_array_name="aux_gfs"
+            )[0]
+
+        # Step 6: Define right-hand sides for the evolution.
+        # Step 6a: uu_rhs = vv - eta_damping*uu:
+        self.uu_rhs = vv - eta_damping * uu
+        # Step 6b: The right-hand side of the \partial_t v equation
+        #          is given by:
+        #          \hat{g}^{ij} \partial_i \partial_j u - \hat{\Gamma}^i \partial_i u.
+        #          ^^^^^^^^^^^^ PART 1 ^^^^^^^^^^^^^^^^ ^^^^^^^^^^ PART 2 ^^^^^^^^^^^
+        self.vv_rhs = sp.sympify(0)
+        for i in range(3):
+            # PART 2:
+            self.vv_rhs -= contractedGammahatU[i] * uu_dD[i]
+            for j in range(3):
+                # PART 1:
+                self.vv_rhs += rfm.ghatUU[i][j] * uu_dDD[i][j]
+
+        # Step 7: Add source term
+        self.vv_rhs += (
+            sp.Rational(1, 8) * ADD_times_AUU * sp.Pow(psi_background + uu, -7)
+        )
+
+        # Step 8: Set residual before multiplying vv_rhs by variable_wavespeed**2
+        self.residual = self.vv_rhs
+
+        # Step 9: Multiply vv_rhs by variable_wavespeed**2 according to the hyperbolization precription
+        self.vv_rhs *= variable_wavespeed * variable_wavespeed
+
```

Comment:
... and here's an example of how you use it:

https://github.com/nrpy/nrpy/blob/6bfedde7517d135545b06930eed09eccdd60754d/nrpy/infrastructures/BHaH/general_relativity/BSSN_C_codegen_library.py#L356

Reply:
This has caused pylint score to drop to 9.07:

``
-={ Step 4: pylint }=-
Pylint score is 9.07
************* Module ConformallyFlat_RHSs
nrpy/equations/nrpyelliptic/ConformallyFlat_RHSs.py:126:53: E1136: Value 'OrderedDict' is unsubscriptable (unsubscriptable-object)
``

I don't understand why, since the code follows very closely the BSSN example.

Reply:
As you can see, it's a Python 3.6.7 issue.

Reply:
Please add this line just below line 78 in your `.pylintrc_python36` file in the root directory of your repo.

```
[tabstop]unsubscriptable-object,     # ADDED: pylint errors on typehints like OrderedDict[str, sp.Expr], but only for some cases!
```

The tabstop is exactly that -- you'll need to press the [tab] button.

Reply:
Done in https://github.com/nrpy/nrpy/pull/17/commits/9e3ce1919ba33abfb3e5c1a7e4cc3899f28b7774.

Reply:
Now, fixed in https://github.com/nrpy/nrpy/pull/17/commits/c8a757d6207978f73970050a8632cb3ee4337f43.